src/senaite/jsonapi/v1/routes/content.py

```diff
@@ -59,26 +59,19 @@ def get(context, request, resource=None, uid=None):
 # http://werkzeug.pocoo.org/docs/0.11/routing/#custom-converters
 @add_route("/<any(" + ACTIONS + "):action>",
            "senaite.jsonapi.v1.action", methods=["POST"])
-@add_route("/<string:resource>",
-           "senaite.jsonapi.v1.action", methods=["POST"])
-@add_route("/<string:resource>/<string(maxlength=32):uid>",
+@add_route("/<string:resource>/<string(length=32):uid>",
            "senaite.jsonapi.v1.action", methods=["POST"])
-@add_route("/<any(" + ACTIONS + "):action>/<string(maxlength=32):uid>",
+@add_route("/<any(" + ACTIONS + "):action>/<string(length=32):uid>",
            "senaite.jsonapi.v1.action", methods=["POST"])
 @add_route("/<string(length=32):uid>",
            "senaite.jsonapi.v1.action", methods=["POST"])
 @add_route("/<string:resource>/<any(" + ACTIONS + "):action>",
            "senaite.jsonapi.v1.action", methods=["POST"])
-@add_route("/<string:resource>/<any(" + ACTIONS + "):action>/<string(maxlength=32):uid>",
+@add_route("/<string:resource>/<any(" + ACTIONS + "):action>/<string(length=32):uid>",
            "senaite.jsonapi.v1.action", methods=["POST"])
 def action(context, request, action=None, resource=None, uid=None):
     """Various HTTP POST actions
     """
-
-    # allow to set the method via the header
-    if action is None:
-        action = request.get_header("HTTP_X_HTTP_METHOD_OVERRIDE", "CREATE").lower()
-
```

Comment:
I think this was needed when adapting something like backbone.js API. The reason is that they map the default update/delete actions to PUT/UPDATE Requests

Reply:
I realized that "create" operation with all params sent via post was working because of this. In fact, when you were calling the operation "update" (with all params sent via post), this condition was evaluating to true and therefore we were ending up with a wrong action ("create"). After I fixed the bug re route masking I decided to not add this conditional again to prevent "undesired" effects and also because I thought (I saw a Pull Request re Plone 5.2 compatibility in `plone.jsonapi.routes`) that this conditional was added as a fix to the same problem I had, but was not addressing the "real" problem. I can add the conditional, but without setting the default to "CREATE".